src/plugins/orangehrmClaimPlugin/Service/ClaimService.php

```diff
@@ -50,4 +56,42 @@ public function getReferenceId(): string
         $date = $this->getDateTimeHelper()->getNow()->format('Ymd');
         return $date . str_pad("$nextId", 7, 0, STR_PAD_LEFT);
     }
+
+    /**
+     * @param int $action
+     * @param ClaimRequest $claimRequest
+     * @return bool
+     */
+    public function isActionAllowed(int $action, ClaimRequest $claimRequest): bool
+    {
+        $isActionAllowed = $this->getUserRoleManager()->isActionAllowed(
+            WorkflowStateMachine::FLOW_CLAIM,
+            $claimRequest->getStatus(),
+            $action,
+            [],
+            [],
+            [Employee::class => $claimRequest->getEmployee()->getEmpNumber()]
+        );
+        if (!$isActionAllowed) {
+            throw $this->getForbiddenException();
+        }
+        return true;
+    }
+
+    /**
+     * @param int $requestId
+     * @return ClaimRequest
+     */
+    public function getClaimRequest(int $requestId): ClaimRequest
+    {
+        $claimRequest = $this->getClaimDao()
+            ->getClaimRequestById($requestId);
+        if (!$claimRequest instanceof ClaimRequest) {
+            throw $this->getInvalidParamException(self::PARAMETER_REQUEST_ID);
+        }
+        if (!$this->getUserRoleManagerHelper()->isEmployeeAccessible($claimRequest->getEmployee()->getEmpNumber())) {
+            throw $this->getForbiddenException();
+        }
+        return $claimRequest;
+    }
```

Comment:
this refactor seems good, but the service is not the right place to move. use trait
ref: `src/plugins/orangehrmLeavePlugin/Api/Traits/LeavePermissionTrait.php` check this